firestore/bulkwriter.go

```diff
@@ -0,0 +1,521 @@
+package firestore
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	vkit "cloud.google.com/go/firestore/apiv1"
+	pb "google.golang.org/genproto/googleapis/firestore/v1"
+)
+
+const (
+	// MaxBatchSize is the max number of writes to send in a request
+	MaxBatchSize = 20
```

Comment:
At least some of this consts, at least this one, seem to be refferenced only internally and are not defaults that a user can change in a way. For such values I would recommend just keeping them private.

Reply:
Good point. I exposed these constants because the Java impl. did the same thing. Having them public isn't a hard requirement (and you're not the first person to point this out). I'll hide them in the next rev.